### ISSUE TITLE
conda run with read-only environment causes infinite loop on windows

### ISSUE THREAD

**Issue by melund** [conda/conda#9574]
## Current Behavior

Using `conda run` on Windows with a environment in a directory requiring admin rights causes an "almost" infinite loop. 

The loop happens when "conda run" [tries to create a ".bat"](https://github.com/conda/conda/blob/master/conda/cli/main_run.py#L22-L23) file with the commands. It tries to write the bat file in the environment prefix but can't

I think the problem is caused by a [bug python std library "tempfile"](https://github.com/python/cpython/blob/master/Lib/tempfile.py#L245-L258), because the loop also happens when using `NamedTemporaryFile` to create a file in directory that requires admin.
```
with NamedTemporaryFile(mode="w", dir="c:\\", delete=False):
    pass
```
(**EDIT:** It is [issue 22107](https://bugs.python.org/issue22107) in CPython)

But don't think we need to wait for python to be fixed. If "conda run" didn't create the ".bat" in the environment prefix but instead in a real tempfile location this would not be a problem. 

Thus, changing [this line](https://github.com/conda/conda/blob/24b41ecbddfe08cc4c41f3a00a841835ecb5d4bb/conda/utils.py#L327)

```
tmp_prefix = abspath(join(prefix, '.tmp'))
```
from a full path to just a normal filename prefix:
```
tmp_prefix = ".conda_subproc_"
```

The [prefix variable in `NamedTemporaryFile`](https://github.com/conda/conda/blob/24b41ecbddfe08cc4c41f3a00a841835ecb5d4bb/conda/utils.py#L340) is not supposed to be used for full paths anyway.

### Steps to Reproduce

1. Create an environment
2. Move a environment to place requiring admin rights for writing E.g. "c:/conda-env" 
3. Use conda run with this environment
```
conda run -p "c:\conda-env" python --version
```

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : C:\Users\mel\Anaconda3
            shell level : 1
       user config file : C:\Users\mel\.condarc
 populated config files : C:\Users\mel\.condarc
          conda version : 4.8.0
    conda-build version : 3.18.11
         python version : 3.7.4.final.0
       virtual packages : __cuda=10.1
       base environment : C:\Users\mel\Anaconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\mel\Anaconda3\pkgs
                          C:\Users\mel\.conda\pkgs
                          C:\Users\mel\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\mel\Anaconda3\envs
                          C:\Users\mel\.conda\envs
                          C:\Users\mel\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.8.0 requests/2.22.0 CPython/3.7.4 Windows/10 Windows/10.0.19041
          administrator : False
             netrc file : None
           offline mode : False

```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> C:\Users\mel\.condarc <==
channels:
  - conda-forge
  - defaults
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at C:\Users\mel\Anaconda3:
#
# Name                    Version                   Build  Channel
_ipyw_jlab_nb_ext_conf    0.1.0                    py37_0    defaults
alabaster                 0.7.12                   py37_0    defaults
anaconda                  2019.10                  py37_0    defaults
anaconda-client           1.7.2                    py37_0    defaults
anaconda-navigator        1.9.7                    py37_0    defaults
anaconda-project          0.8.3                      py_0    conda-forge
anypytools                1.2.1                    pypi_0    pypi
apipkg                    1.5                        py_0    conda-forge
appdirs                   1.4.3                      py_1    conda-forge
arrow                     0.15.5                   py37_0    conda-forge
asn1crypto                1.0.1                    py37_0    conda-forge
astroid                   2.3.1                    py37_0    conda-forge
astropy                   3.2.1            py37he774522_0    defaults
atomicwrites              1.3.0                    py37_1    defaults
attrs                     19.2.0                     py_0    conda-forge
babel                     2.7.0                      py_0    conda-forge
backcall                  0.1.0                    py37_0    defaults
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
backports.os              0.1.1                    py37_0    defaults
backports.shutil_get_terminal_size 1.0.0                    py37_2    defaults
backports.tempfile        1.0                        py_0    conda-forge
backports.weakref         1.0.post1             py37_1000    conda-forge
backtrace                 0.2.1                    pypi_0    pypi
beautifulsoup4            4.8.0                    py37_0    conda-forge
binaryornot               0.4.4                      py_1    conda-forge
bitarray                  1.0.1            py37he774522_0    defaults
bkcharts                  0.2                      py37_0    defaults
black                     19.3b0                     py_0    conda-forge
blas                      1.0                         mkl    defaults
bleach                    3.1.0                    py37_0    defaults
blosc                     1.16.3               h7bd577a_0    defaults
bokeh                     1.3.4                    py37_0    conda-forge
boto                      2.49.0                   py37_0    defaults
bottleneck                1.2.1            py37h452e1ab_1    defaults
btk                       0.4.dev0         py37ha925a31_0    anybody
bzip2                     1.0.8                he774522_0    defaults
ca-certificates           2019.8.28                     0    defaults
certifi                   2019.9.11                py37_0    conda-forge
cffi                      1.12.3           py37h7a1dbc1_0    defaults
chardet                   3.0.4                 py37_1003    conda-forge
click                     7.0                      py37_0    defaults
cloudpickle               1.2.2                      py_0    conda-forge
clyent                    1.2.2                    py37_1    defaults
cmarkgfm                  0.4.2            py37hfa6e2cd_2    conda-forge
colorama                  0.3.7                    pypi_0    pypi
comtypes                  1.1.7                    py37_0    defaults
conda                     4.8.0                    py37_1    conda-forge
conda-build               3.18.11                  py37_0    conda-forge
conda-env                 2.6.0                         1    defaults
conda-package-handling    1.6.0            py37h2fa13f4_1    conda-forge
conda-verify              3.4.2                      py_1    defaults
configargparse            0.15.2                   pypi_0    pypi
console_shortcut          0.1.1                         3    defaults
contextlib2               0.6.0                      py_0    conda-forge
cookiecutter              1.7.0                      py_0    conda-forge
cookies                   2.2.1                      py_0    conda-forge
cryptography              2.7              py37h7a1dbc1_0    defaults
curl                      7.65.3               h2a8f88b_0    defaults
cycler                    0.10.0                   py37_0    defaults
cython                    0.29.13          py37ha925a31_0    defaults
cytoolz                   0.10.0           py37he774522_0    defaults
dask                      2.5.2                      py_0    conda-forge
dask-core                 2.5.2                      py_0    conda-forge
datrie                    0.8                      pypi_0    pypi
decorator                 4.4.0                    py37_1    defaults
defusedxml                0.6.0                      py_0    conda-forge
distributed               2.5.2                      py_0    conda-forge
docutils                  0.15.2                   py37_0    conda-forge
entrypoints               0.3                      py37_0    defaults
et_xmlfile                1.0.1                    py37_0    defaults
execnet                   1.7.1                      py_0    conda-forge
ezc3d                     1.2.2            py37h79cbd7a_0    conda-forge
fastcache                 1.1.0            py37he774522_0    defaults
ffmpeg                    4.2                  h6538335_0    conda-forge
filelock                  3.0.12                     py_0    defaults
flask                     1.1.1                      py_0    conda-forge
freetype                  2.9.1                ha9979f8_1    defaults
fsspec                    0.5.2                      py_0    conda-forge
future                    0.18.2                   py37_0    conda-forge
get_terminal_size         1.0.0                h38e98db_0    defaults
gevent                    1.4.0            py37he774522_0    defaults
gitdb2                    2.0.6                    pypi_0    pypi
gitpython                 3.0.5                    pypi_0    pypi
glob2                     0.7                        py_0    conda-forge
greenlet                  0.4.15           py37hfa6e2cd_0    conda-forge
h5py                      2.9.0            py37h5e291fa_0    defaults
hdf5                      1.10.4               h7ebc959_0    defaults
heapdict                  1.0.1                      py_0    conda-forge
html5lib                  1.0.1                    py37_0    defaults
icc_rt                    2019.0.0             h0cc432a_1    defaults
icu                       58.2                 ha66f8fd_1    defaults
idna                      2.8                      py37_0    defaults
imageio                   2.6.0                    py37_0    conda-forge
imagesize                 1.1.0                    py37_0    defaults
importlib_metadata        0.23                     py37_0    conda-forge
intel-openmp              2019.4                      245    defaults
ipykernel                 5.1.2            py37h39e3cac_0    defaults
ipython                   7.8.0            py37h39e3cac_0    defaults
ipython_genutils          0.2.0                    py37_0    defaults
ipywidgets                7.5.1                      py_0    conda-forge
isort                     4.3.21                   py37_0    conda-forge
itsdangerous              1.1.0                    py37_0    defaults
jdcal                     1.4.1                      py_0    conda-forge
jedi                      0.15.1                   py37_0    conda-forge
jinja2                    2.10.3                     py_0    conda-forge
jinja2-time               0.2.0                      py_2    conda-forge
joblib                    0.13.2                   py37_0    defaults
jom                       1.1.3                         0    conda-forge
jpeg                      9b                   hb83a4c4_2    defaults
json5                     0.8.5                      py_0    conda-forge
jsonschema                3.0.2                    py37_0    conda-forge
jupyter                   1.0.0                    py37_7    defaults
jupyter_client            5.3.3                    py37_1    conda-forge
jupyter_console           6.0.0                    py37_0    defaults
jupyter_core              4.5.0                      py_0    conda-forge
jupyterlab                1.1.4              pyhf63ae98_0    defaults
jupyterlab_server         1.0.6                      py_0    conda-forge
keyring                   18.0.0                   py37_0    defaults
kiwisolver                1.1.0            py37ha925a31_0    defaults
krb5                      1.16.1               hc04afaa_7    defaults
lazy-object-proxy         1.4.2            py37he774522_0    defaults
libarchive                3.3.3                h0643e63_5    defaults
libcurl                   7.65.3               h2a8f88b_0    defaults
libiconv                  1.15                 h1df5818_7    defaults
liblief                   0.9.0                ha925a31_2    defaults
libpng                    1.6.37               h2a8f88b_0    defaults
libsodium                 1.0.16               h9d3ae62_0    defaults
libssh2                   1.8.2                h7a1dbc1_0    defaults
libtiff                   4.0.10               hb898794_2    defaults
libxml2                   2.9.9                h464c3ec_0    defaults
libxslt                   1.1.33               h579f668_0    defaults
llvmlite                  0.29.0           py37ha925a31_0    defaults
locket                    0.2.0                    py37_1    defaults
lxml                      4.4.1            py37h1350720_0    defaults
lz4-c                     1.8.1.2              h2fa13f4_0    defaults
lzo                       2.10                 h6df0209_2    defaults
m2w64-gcc-libgfortran     5.3.0                         6    defaults
m2w64-gcc-libs            5.3.0                         7    defaults
m2w64-gcc-libs-core       5.3.0                         7    defaults
m2w64-gmp                 6.1.0                         2    defaults
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    defaults
markupsafe                1.1.1            py37he774522_0    defaults
matplotlib                3.1.1            py37hc8f65d3_0    defaults
mccabe                    0.6.1                    py37_1    defaults
menuinst                  1.4.16           py37he774522_0    defaults
mistune                   0.8.4            py37he774522_0    defaults
mkl                       2019.4                      245    defaults
mkl-service               2.3.0            py37hb782905_0    defaults
mkl_fft                   1.0.14           py37h14836fe_0    defaults
mkl_random                1.1.0            py37h675688f_0    defaults
mock                      3.0.5                    py37_0    conda-forge
more-itertools            7.2.0                    py37_0    defaults
mpmath                    1.1.0                    py37_0    defaults
msgpack-python            0.6.1            py37h74a9793_1    defaults
msys2-conda-epoch         20160418                      1    defaults
multipledispatch          0.6.0                    py37_0    defaults
navigator-updater         0.2.1                    py37_0    defaults
nbconvert                 5.6.0                    py37_1    conda-forge
nbformat                  4.4.0                    py37_0    defaults
networkx                  2.3                        py_0    conda-forge
nltk                      3.4.5                    py37_0    defaults
nodejs                    12.8.1                        0    conda-forge
nose                      1.3.7                    py37_2    conda-forge
notebook                  6.0.1                    py37_0    conda-forge
numba                     0.45.1           py37hf9181ef_0    defaults
numexpr                   2.7.0            py37hdce8814_0    defaults
numpy                     1.16.5           py37h19fb1c0_0    defaults
numpy-base                1.16.5           py37hc3f5095_0    defaults
numpydoc                  0.9.1                      py_0    conda-forge
olefile                   0.46                     py37_0    defaults
openpyxl                  3.0.0                      py_0    conda-forge
openssl                   1.1.1d               he774522_2    defaults
packaging                 19.2                       py_0    conda-forge
pandas                    0.25.1           py37ha925a31_0    defaults
pandoc                    2.2.3.2                       0    defaults
pandocfilters             1.4.2                    py37_1    defaults
parso                     0.5.1                      py_0    conda-forge
partd                     1.0.0                      py_0    conda-forge
path.py                   12.0.1                     py_0    conda-forge
pathlib2                  2.3.5                    py37_0    conda-forge
patsy                     0.5.1                    py37_0    defaults
pep8                      1.7.1                    py37_0    defaults
pickleshare               0.7.5                    py37_0    defaults
pillow                    6.2.0            py37hdc69c19_0    defaults
pip                       19.2.3                   py37_0    conda-forge
pkginfo                   1.5.0.1                  py37_0    defaults
pluggy                    0.13.0                   py37_0    conda-forge
ply                       3.11                     py37_0    defaults
powershell_shortcut       0.0.1                         2    defaults
poyo                      0.5.0                      py_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt-toolkit            3.0.0                    pypi_0    pypi
prompt_toolkit            2.0.10                     py_0    conda-forge
psutil                    5.6.3            py37he774522_0    defaults
ptvsd                     4.3.2            py37hfa6e2cd_0    conda-forge
py                        1.8.0                    py37_0    defaults
py-lief                   0.9.0            py37ha925a31_2    defaults
pycodestyle               2.5.0                    py37_0    defaults
pycosat                   0.6.3            py37hfa6e2cd_0    defaults
pycparser                 2.19                     py37_0    defaults
pycrypto                  2.6.1            py37hfa6e2cd_9    defaults
pycurl                    7.43.0.3         py37h7a1dbc1_0    defaults
pydoe                     0.3.8                      py_1    conda-forge
pyflakes                  2.1.1                    py37_0    defaults
pygments                  2.4.2                      py_0    conda-forge
pygments_anyscript        1.2.1                      py_2    conda-forge
pylint                    2.4.2                    py37_0    conda-forge
pyodbc                    4.0.27           py37ha925a31_0    defaults
pyopenssl                 19.0.0                   py37_0    conda-forge
pyparsing                 2.4.2                      py_0    conda-forge
pyqt                      5.9.2            py37h6538335_2    conda-forge
pyreadline                2.1                      py37_1    defaults
pyrsistent                0.15.4           py37he774522_0    defaults
pysocks                   1.7.1                    py37_0    conda-forge
pytables                  3.5.2            py37h1da0976_1    defaults
pytest                    5.2.1                    py37_0    conda-forge
pytest-arraydiff          0.3              py37h39e3cac_0    defaults
pytest-astropy            0.5.0                    py37_0    defaults
pytest-doctestplus        0.4.0                      py_0    conda-forge
pytest-forked             1.1.2                      py_0    conda-forge
pytest-openfiles          0.4.0                      py_0    conda-forge
pytest-remotedata         0.3.2                    py37_0    defaults
pytest-xdist              1.30.0                     py_0    conda-forge
python                    3.7.4                h5263a28_0    defaults
python-dateutil           2.8.0                    py37_0    defaults
python-libarchive-c       2.8                     py37_13    defaults
pytz                      2019.3                     py_0    conda-forge
pywavelets                1.0.3            py37h8c2d366_1    defaults
pywin32                   223              py37hfa6e2cd_1    defaults
pywinpty                  0.5.5                 py37_1000    conda-forge
pyyaml                    5.1.2            py37he774522_0    defaults
pyzmq                     18.1.0           py37ha925a31_0    defaults
qt                        5.9.7            vc14h73c81de_0    defaults
qtawesome                 0.6.0                      py_0    conda-forge
qtconsole                 4.5.5                      py_0    conda-forge
qtpy                      1.9.0                      py_0    conda-forge
ratelimiter               1.2.0.post0              pypi_0    pypi
readme_renderer           24.0                       py_0    conda-forge
requests                  2.22.0                   py37_0    conda-forge
requests-toolbelt         0.9.1                      py_0    conda-forge
responses                 0.9.0                      py_0    conda-forge
ripgrep                   11.0.2               h301d43c_3    conda-forge
rope                      0.14.0                     py_0    conda-forge
rsdbxlib                  1.0                       dev_0    <develop>
ruamel_yaml               0.15.46          py37hfa6e2cd_0    defaults
scikit-image              0.15.0           py37ha925a31_0    defaults
scikit-learn              0.21.3           py37h6288b17_0    defaults
scipy                     1.3.1            py37h29ff71c_0    conda-forge
seaborn                   0.9.0                    py37_0    defaults
send2trash                1.5.0                    py37_0    defaults
setproctitle              1.1.10          py37hfa6e2cd_1001    conda-forge
setuptools                41.4.0                   py37_0    conda-forge
simplegeneric             0.8.1                    py37_2    defaults
singledispatch            3.4.0.3                  py37_0    defaults
sip                       4.19.8           py37h6538335_0    conda-forge
six                       1.12.0                   py37_0    defaults
smmap2                    2.0.5                    pypi_0    pypi
snakemake                 5.8.1+62.gdb24a58b           dev_0    <develop>
snappy                    1.1.7                h777316e_3    defaults
snowballstemmer           2.0.0                      py_0    conda-forge
sortedcollections         1.1.2                    py37_0    defaults
sortedcontainers          2.1.0                    py37_0    defaults
soupsieve                 1.9.3                    py37_0    conda-forge
sphinx                    2.2.0                      py_0    conda-forge
sphinxcontrib             1.0                      py37_1    defaults
sphinxcontrib-applehelp   1.0.1                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.1                      py_0    conda-forge
sphinxcontrib-htmlhelp    1.0.2                      py_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.2                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.3                      py_0    conda-forge
sphinxcontrib-websupport  1.1.2                      py_0    conda-forge
spyder                    3.3.6                    py37_0    conda-forge
spyder-kernels            0.5.2                    py37_0    conda-forge
sqlalchemy                1.3.9            py37he774522_0    defaults
sqlite                    3.30.0               he774522_0    defaults
statsmodels               0.10.1           py37h8c2d366_0    defaults
swig                      4.0.1                         0    conda-forge
sympy                     1.4                      py37_0    conda-forge
tbb                       2019.4               h74a9793_0    defaults
tblib                     1.4.0                      py_0    conda-forge
terminado                 0.8.2                    py37_0    conda-forge
testpath                  0.4.2                    py37_0    defaults
tk                        8.6.8                hfa6e2cd_0    conda-forge
toml                      0.10.0                     py_0    conda-forge
toolz                     0.10.0                     py_0    conda-forge
tornado                   6.0.3            py37he774522_0    defaults
tqdm                      4.36.1                     py_0    conda-forge
traitlets                 4.3.3                    py37_0    conda-forge
twine                     1.13.0                     py_1    conda-forge
unicodecsv                0.14.1                   py37_0    defaults
urllib3                   1.24.2                   py37_0    defaults
vc                        14.1                 h0510ff6_4    defaults
vitables                  3.0.0            py37hfe2a811_5    conda-forge
vs2015_runtime            14.16.27012          hf0eaf9b_0    defaults
wcwidth                   0.1.7                    py37_0    defaults
webencodings              0.5.1                    py37_1    defaults
werkzeug                  0.16.0                     py_0    conda-forge
wheel                     0.33.6                   py37_0    conda-forge
whichcraft                0.6.1                      py_0    conda-forge
widgetsnbextension        3.5.1                    py37_0    conda-forge
win_inet_pton             1.1.0                    py37_0    conda-forge
win_unicode_console       0.5                      py37_0    conda-forge
wincertstore              0.2                      py37_0    defaults
winpty                    0.4.3                         4    conda-forge
wrapt                     1.11.2           py37he774522_0    defaults
xlrd                      1.2.0                    py37_0    defaults
xlsxwriter                1.2.1                      py_0    conda-forge
xlwings                   0.15.10                  py37_0    conda-forge
xlwt                      1.3.0                    py37_0    defaults
xmltodict                 0.12.0                     py_0    conda-forge
xonsh                     0.9.13                    dev_0    <develop>
xontrib-readable-traceback 0.3.1                    pypi_0    pypi
xz                        5.2.4                h2fa13f4_4    defaults
yaml                      0.1.7                hc54c509_2    defaults
zeromq                    4.3.1                h33f27b4_3    defaults
zict                      1.0.0                      py_0    conda-forge
zipp                      0.6.0                      py_0    conda-forge
zlib                      1.2.11               h62dcd97_3    defaults
zstd                      1.3.7                h508b16e_0    defaults

```
</p></details>

**Comment by melund** [comment]
This bug causes VS Code to launch hanging python process which use max CPU if there are any conda environments on the machine in places that require admin rights to access. 

This is because Microsoft Python extension to VS code uses `conda run` to inspect any conda environments on the machine.

**Comment by jezdez** [comment]
This is very related to #10098 and #11208 and has been fixed in #11215.
